### PR TITLE
Personal prices for klasses

### LINKED ISSUE #139: Personal Pricing
As an application reviewer, I would like to be able to personalize applicants tuition price. This would allow for discounts and scholarships. 

The FluidReview application can be configured to add a price field to the submission metadata. When an applicant is accepted, the price field can be included in the webhook call to bootcamp-ecommerce. 

### acceptance criteria
- [ ] add a personal price field, perhaps to the `bootcampadmissioncache` model
- [ ] override the amount field in the installment model with this personal price
  (note that we are no longer planning to set up multiple installments for payment)

### PR DESCRIPTION
**Pull request by mbertrand**
#### What are the relevant tickets?
Closes #139

#### What's this PR do?
Adds a `PersonalPrice` model to track user-specific pricing for a klass, and modifies klass price lookups to query this model first before getting the default price.

This PR assumes that the `amount_to_pay` field in a `WebhookRequest` is the personal user price for a klass. 

#### How should this be manually tested?
- Create a `Bootcamp`, `Klass` (`klass_key`=72713), and `Installment`.
- Send this `WebhookRequest` request, with `amount_to_pay` blank:
  ```curl -H 'Authorization: Basic faketoken' -H 'Content-type: text/plain' -d '{"date_of_birth":"","user_email":"staff@example.com","amount_to_pay":"","user_id":95292895,"submission_id":4621059,"award_id":72713,"award_cost":"1000","award_name":"TESTCAMP"}' http://192.168.99.101:8099/api/v0/fluidreview_webhook/```
- Log in via local EdX as `staff@example.com`
- Go to the payment page for the klass.  You should owe the same amount as you specified for the klass installment.
- Send the WebHookRequest again with `amount_to_pay` set to `'999.99'`.  Then reload the page. It should now say you owe $999.99 for the class.
- Send the WebHookRequest again with a different amount.  Then reload the page, you should see the new amount owed.
- Send the WebHookRequest again with a blank amount.  Then reload the page, you should see the default klass installment amount again.

### CODE REVIEW

File: fluidreview/api.py [hunk 1]
```diff
@@ -207,18 +209,29 @@ def parse_webhook(webhook):
...
         # Get user info from FluidReview API (ensures that user id is real).
-        user_info = FluidReviewAPI().get('/users/{}'.format(fluidreview_id)).json()
+        user_info = FluidReviewAPI().get('/users/{}'.format(webhook.user_id)).json()
         serializer = UserSerializer(data=user_info)
         serializer.is_valid(raise_exception=True)
-        process_user(serializer.data)
+        user = process_user(serializer.data)
+    else:
+        user = profile.user
+    if webhook.amount_to_pay:
```
Comment (noisecapella) [inline fluidreview/api.py]: Should this be `webhook.amount_to_pay is not None`?

File: fluidreview/api.py [hunk 2]
```diff
@@ -207,18 +209,29 @@ def parse_webhook(webhook):
...
-        fluidreview_id(int): ID of FluidReview user
+        webhook(WebhookRequest): a WebHookRequest object
     """
-    if not Profile.objects.filter(fluidreview_id=fluidreview_id):
+    klass = Klass.objects.get(klass_key=webhook.award_id)
+    profile = Profile.objects.filter(fluidreview_id=webhook.user_id).first()
+    if not profile:
         # Get user info from FluidReview API (ensures that user id is real).
-        user_info = FluidReviewAPI().get('/users/{}'.format(fluidreview_id)).json()
+        user_info = FluidReviewAPI().get('/users/{}'.format(webhook.user_id)).json()
```
Comment (noisecapella) [inline fluidreview/api.py]: Can you add a check that `webhook.user_id` is not `None`?

File: klasses/models.py [hunk 3]
```diff
@@ -112,6 +113,21 @@ def total_due_by_next_deadline(self):
...
+        Returns the personal price (if any) or standard price for a klass
+
+        Args:
+            user(User): the user to get a price for
+
+        Returns:
+            Decimal: the price for the klass
+        """
+        personal_price = self.personal_prices.filter(user=user).first()
+        if personal_price:
```
Comment (noisecapella) [inline klasses/models.py]: Can you change this to `personal_price is not None`?

File: fluidreview/api.py (hunk 2 above)
Comment (mbertrand) [reply fluidreview/api.py]: ✅

File: fluidreview/api.py [hunk 4]
```diff
@@ -175,6 +176,7 @@ def process_user(fluid_user):
         profile.fluidreview_id = fluid_user['id']
         profile.name = fluid_user['full_name']
         profile.save()
+    return user
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply fluidreview/api.py]: ✅

File: fluidreview/api.py (hunk 1 above)
Comment (mbertrand) [reply fluidreview/api.py]: ✅

File: klasses/models.py (hunk 3 above)
Comment (mbertrand) [reply klasses/models.py]: ✅

**Pull request merged**